pkg/controller/postgresqldatabase/postgresqldatabase_controller.go

```diff
@@ -0,0 +1,235 @@
+	log.Info(fmt.Sprintf("Creating service user with name %s and password %s", name, password))
+
+	// Create the service user
+	_, err := r.db.Exec(fmt.Sprintf("CREATE USER %s WITH PASSWORD '%s' NOCREATEROLE VALID UNTIL 'infinity'", name, password))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_object" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Service user; %s already exists", name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Service user; %s created", name))
+	}
+
+	// Create the database
+	_, err = r.db.Exec(fmt.Sprintf("CREATE DATABASE %s", name))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_database" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Database; %s already exists", name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Database; %s created", name))
+	}
+
+	// Alter ownership of the database to the database user
+	// TODO: Handle the error in a better way
+	_, err = r.db.Exec(fmt.Sprintf("ALTER DATABASE %s OWNER TO %s", name, name))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_object" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Database; %s already exists", name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Granted %s owner of the database; %s", name, name))
+	}
+
+	serviceConnection, err := postgresqlConnection(fmt.Sprintf("postgresql://%s:%s@localhost:5432/%s?sslmode=disable", name, password, name))
+	if err != nil {
+		return err
+	}
+
+	// Create schema in the database
+	_, err = serviceConnection.Exec(fmt.Sprintf("CREATE SCHEMA %s", name))
+	if err != nil {
+		pqError, ok := err.(*pq.Error)
+		if !ok || pqError.Code.Name() != "duplicate_schema" {
+			return err
+		}
+		log.Info(fmt.Sprintf("Schema; %s already exists in database; %s", name, name), "errorCode", pqError.Code, "errorName", pqError.Code.Name())
+	} else {
+		log.Info(fmt.Sprintf("Schema; %s created in database; %s", name, name))
+	}
+	return nil
+}
+
+func (r *ReconcilePostgreSQLDatabase) getSecretValue(name, namespace, key string) (string, error) {
+	secret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, secret)
+	if err != nil {
+		return "", err
+	}
+
+	password, err := base64.StdEncoding.DecodeString(string(secret.Data[key]))
+	if err != nil {
+		return "", err
+	}
+	return string(password), nil
+}
+
+func (r *ReconcilePostgreSQLDatabase) getConfigMapValue(name, namespace, key string) (string, error) {
+	configMap := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, configMap)
+	if err != nil {
+		return "", err
+	}
+	return string(configMap.Data[key]), nil
```

Comment:
And again same as above. If we use a non-existing key we just get an empty value. Would be nice if the controller did not accept that condition as to inform the users that they referenced the wrong key instead of just using an empty password.